src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -602,7 +602,7 @@ private void linkToParents(ResultSet rs, ResultMapping parentMapping, Object row
     CacheKey parentKey = createKeyForMultipleResults(rs, parentMapping, parentMapping.getColumn(),
         parentMapping.getForeignColumn());
     List<PendingRelation> parents = pendingRelations.get(parentKey);
-    if (parents != null) {
+    if (parents != null && !parents.isEmpty()) {
```

Comment:
I am not sure why this is necessary or if it is valid.
Please add a test case that justifies the change.